tabular/error.go

```diff
@@ -0,0 +1,30 @@
+package tabular
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type JSONErrorResponse struct {
+	Error JSONError `json:"error"`
+}
+
+type JSONError struct {
+	Message string `json:"message"`
+	Type    string `json:"type"`
+	Code    int    `json:"code"`
+}
+
+func ParseErrorResponse(response io.ReadCloser) (JSONErrorResponse, error) {
+	var errResp JSONErrorResponse
+	body, err := io.ReadAll(response)
+	if err != nil {
+		return JSONErrorResponse{}, err
```

Comment:
Does the actual error get entirely swallowed if the JSON fails to read/parse? Or just uglier?

Reply:
The caller would have to decide what to do in that case. It will still get an error about trying to read the response.